packages/mobile-hooks/src/messages/useSync.ts

```diff
@@ -732,34 +736,39 @@ export const useSync = (onSync?: () => Promise<any>) => {
-
-                  const delDocs = documents
-                    .filter(
-                      (d) =>
-                        (d.status === 'PROCESSED' || d.status === 'ARCHIVE') &&
-                        new Date(d.documentDate).getTime() <= maxDocDate.getTime(),
-                    )
-                    .map((d) => d.id);
-
-                  if (delDocs.length) {
-                    addRequestNotice(
-                      `Удаление обработанных документов, дата которых менее ${getDateString(maxDocDate)}`,
-                    );
+                const removeDocumentsByStatus = async (status: StatusType, cleanTime: number, message: string) => {
+                  if (cleanTime > 0) {
+                    const maxDocDate = new Date();
+                    maxDocDate.setDate(maxDocDate.getDate() - cleanTime);
 
-                    const delDocResponse = await docDispatch(documentActions.removeDocuments(delDocs));
-                    if (delDocResponse.type === 'DOCUMENTS/REMOVE_MANY_FAILURE') {
-                      addError(
-                        'useSync: removeDocuments',
-                        `Обработанные документы, дата которых менее ${getDateString(maxDocDate)}, не удалены`,
-                        tempErrs,
+                    const delDocs = documents
+                      .filter((d) => d.status === status && new Date(d.documentDate).getTime() <= maxDocDate.getTime())
+                      .map((d) => d.id);
+
+                    if (delDocs.length) {
+                      addRequestNotice(
+                        `Удаление документов со статусом "${message}", дата которых менее ${getDateString(maxDocDate)}`,
                       );
+
+                      const delDocResponse = await docDispatch(documentActions.removeDocuments(delDocs));
+                      if (delDocResponse.type === 'DOCUMENTS/REMOVE_MANY_FAILURE') {
+                        addError(
+                          'useSync: removeDocuments',
+                          `Документы со статусом "${message}", дата которых менее ${getDateString(
+                            maxDocDate,
+                          )}, не удалены`,
+                          tempErrs,
+                        );
+                      }
                     }
                   }
-                }
+                };
+
+                removeDocumentsByStatus('SENT', cleanSentDocTime, 'Отправленные');
+                removeDocumentsByStatus('DRAFT', cleanDraftDocTime, 'Черновик');
+                removeDocumentsByStatus('READY', cleanReadyDocTime, 'Готов');
+                removeDocumentsByStatus('PROCESSED', cleanDocTime, 'Обработанные');
+                removeDocumentsByStatus('ARCHIVE', cleanDocTime, 'Обработанные');
 
                 if (!connectError && needRequest(syncRequests, 'GET_USER_SETTINGS', currentDate)) {
                   addRequestNotice('Запрос настроек пользователя');
```

Comment:
The code patch seems to introduce new functionality to handle document cleaning based on different status types. Several new constants are added for clean times related to specific document statuses. A new 'removeDocumentsByStatus' function is defined to encapsulate the logic of filtering and removing documents based on the provided status, clean time, and a message describing the status. The function is then called with different status types ('SENT', 'DRAFT', 'READY', 'PROCESSED', 'ARCHIVE') and their respective clean time values to remove documents accordingly.

Quality Rate: 8/10